File: RouteMapping.Server/Models/Route.cs
namespace RouteMapping.Server.Models
{
    public class Route
    {
        public int Id { get; set; }
        public string StartPoint { get; set; }
        public List<Waypoint> Waypoints { get; set; }
        public TimeSpan TotalTime { get; set; }
        public double Distance { get; set; }
    }
}

File: routemapping.client/src/app/models/route.ts
export interface Route {
  id: number;
  startPoint: string;
  waypoints: Waypoint[];
  totalTime: string;
  distance: number;
}

export interface Waypoint {
  id: number;
  routeId: number;
  locationName: string;
  latitude: number;
  longitude: number;
  timeFromStart: string;
}
